Added individual pages for each data type. (#543)
Codecademy/docs: content/general/concepts/boolean/boolean.md

---

> Title: 'Booleans'

Reviewer: ```suggestion
Title: 'Boolean'
```

> Description: 'The boolean data type.'

Reviewer: ```suggestion
Description: 'The boolean data type represents the logical values of true and false.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Data Types'
  - 'Types'
CatalogContent:
  - 'paths/code-foundations'
  - 'paths/computer-science'
---

> Booleans hold a true or false value. Booleans are used in conditional statements. Ex:

Reviewer: ```suggestion
Booleans hold a true or false value and are mostly used in conditional statements to control a program's flow of execution. Some languages use comparison operators (`>=`, `==`, `is`, `not`, etc.) to produce boolean values, while others may use "falsy" values like `""`, `null`, or `0`. Some even use a combination of both of these approaches.
```

> 

Reviewer: ```suggestion

## JavaScript Example

```

> ```javascript

Reviewer: ```suggestion
```js
```

> if (true) {
>   // true would always evaluate to true so the folloing line will always run.
>   console.log('this will always print');
> } else if (false) {
>   // false will alawys evaluate to false so the following lines will never run
>   console.log('This will never print');
> }

Reviewer: ```suggestion
if (42 !== "42") {
  // If the statement above always evaluates to true...
  console.log('This will always print.');
} else {
  console.log('This will never print.');
};
```

---
Title: 'Boolean'
Description: 'The boolean data type represents the logical values of true and false.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Data Types'
  - 'Types'
CatalogContent:
  - 'paths/code-foundations'
  - 'paths/computer-science'
---

Booleans hold a true or false value and are mostly used in conditional statements to control a program's flow of execution. This data type is named after George Boole, a 19th century English logician.

Some languages use comparison operators (`>=`, `==`, `is`, `not`, etc.) to produce boolean values, while others may use "falsy" values like `""`, `null`, or `0`. Some even use a combination of both of these approaches.

## JavaScript Example

> The following example uses [a JavaScript conditional statement](https://www.codecademy.com/resources/docs/javascript/conditionals), along with a comparison [operator](https://www.codecademy.com/resources/docs/javascript/operators) to return `true` or `false` if 42 is, in fact, "42":

Reviewer 2: ```suggestion
The following example uses a comparison [operator](https://www.codecademy.com/resources/docs/javascript/operators) to return `true` or `false` based on the evaluation of the `42 != 5` statement. Then, the [JavaScript conditional](https://www.codecademy.com/resources/docs/javascript/conditionals) determines whether the `if` or `else` block will run.
```

```codebyte/javascript

> if (42 !== "42") {

Author: I think we should change this so it's 2 different numbers so it's a little less confusing to beginners. What do you think

Reviewer: What if we tried this, instead?

```codebyte/javascript
if(4 != 5) {
  // If the statement above always evaluates to true...
  console.log('This will always print.');
} else {
  console.log('This will never print.');
};
```

Instead of using strict comparisons like `!==`, let's just use two different values (of the same data type) and use a regular comparison operator (`!=`). The code should still evaluate to `true` but we'd have to edit the text above in line # 21 to reflect this change.

Author: Yea. Make it simpler because a beginner might not understand the strict comparison and why `42 !== "42"`  evaluates to false.

Reviewer: Absolutely. Strict comparison can be a bit much at first. And fwiw, `42 !== "42"` evaluates to `true` 😄

Author: @Dusch4593 And sometimes people that know this make a mistake. (me 😂)

---
Title: 'Boolean'
Description: 'The boolean data type represents the logical values of true and false.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Data Types'
  - 'Types'
CatalogContent:
  - 'paths/code-foundations'
  - 'paths/computer-science'
---

Booleans hold a true or false value and are mostly used in conditional statements to control a program's flow of execution. This data type is named after George Boole, a 19th century English logician.

Some languages use comparison operators (`>=`, `==`, `is`, `not`, etc.) to produce boolean values, while others may use "falsy" values like `""`, `null`, or `0`. Some even use a combination of both of these approaches.

> Because functions are derived from objects, their closest classification is with composite data types.

Reviewer: @ethan-master-coding Was this supposed to be with `function.md`?

Author: @Dusch4593 I think only line 19. Let me move that now

## JavaScript Example

> The following example uses [a JavaScript conditional statement](https://www.codecademy.com/resources/docs/javascript/conditionals), along with a comparison [operator](https://www.codecademy.com/resources/docs/javascript/operators) to return `true` or `false` if 42 is, in fact, "42":

Reviewer: ```suggestion
The following example uses [a JavaScript conditional statement](https://www.codecademy.com/resources/docs/javascript/conditionals), along with a comparison [operator](https://www.codecademy.com/resources/docs/javascript/operators) to return `true` or `false` if 42 is, in fact, "42":

```
